source/lac/scalapack.cc

```diff
@@ -263,6 +263,131 @@ ScaLAPACKMatrix<NumberType>::operator=(const FullMatrix<NumberType> &matrix)
 
 
 
+template <typename NumberType>
+void
+ScaLAPACKMatrix<NumberType>::copy_from(const LAPACKFullMatrix<NumberType> &B,
+                                       const unsigned int                  r)
+{
+  if (n_rows * n_columns == 0)
+    return;
+
+  const unsigned int this_mpi_process(
+    Utilities::MPI::this_mpi_process(this->grid->mpi_communicator));
+
+  // root process has to be active in the grid of A
+  if (this_mpi_process == r)
+    {
+      Assert(grid->mpi_process_is_active, ExcInternalError());
+      Assert(n_rows == int(B.m()), ExcDimensionMismatch(n_rows, B.m()));
+      Assert(n_columns == int(B.n()), ExcDimensionMismatch(n_columns, B.n()));
+    }
+  // Create 1x1 grid for matrix B.
```

Comment:
Maybe add a comment that u copy via Scalapack functions and thus need grid.

And a brief summary of the algorithm